Cargo.toml

```diff
@@ -422,8 +422,10 @@ macaddr = { version = "1.0.1", features = ["serde_std"] }
 maplit = "1.0.2"
 mockall = "0.13"
 newtype_derive = "0.1.6"
-mg-admin-client = { git = "https://github.com/oxidecomputer/maghemite", rev = "9e0fe45ca3862176dc31ad8cc83f605f8a7e1a42" }
-ddm-admin-client = { git = "https://github.com/oxidecomputer/maghemite", rev = "9e0fe45ca3862176dc31ad8cc83f605f8a7e1a42" }
+# mg-admin-client = { git = "https://github.com/oxidecomputer/maghemite", rev = "9e0fe45ca3862176dc31ad8cc83f605f8a7e1a42" }
+# ddm-admin-client = { git = "https://github.com/oxidecomputer/maghemite", rev = "9e0fe45ca3862176dc31ad8cc83f605f8a7e1a42" }
+mg-admin-client = { git = "https://github.com/oxidecomputer/maghemite", branch = "hyper-v1-no-merge" }
+ddm-admin-client = { git = "https://github.com/oxidecomputer/maghemite", branch = "hyper-v1-no-merge" }
```

Comment:
The explanation for what we're doing here makes sense (there's a circular dependency between maghemite and omicron via omicron-common, and the maghemite `main` branch is broken with respect to omicron APIs) but this is also some hackery of a pretty high order.

I *think* it makes sense -- it moves maghemite forward as much as we can, although it *is* relying on a random ahl-owned branch, rather than something merged -- but it also merits some pretty clear explanation, outside of digging through git history.

Could we file an issue describing what's happening here, and how to get ourselves out of this situation?